tests/CMakeLists.txt

```diff
@@ -74,17 +92,35 @@ set(OPENSHOT_TEST_FILES
   KeyFrame_Tests.cpp
   Point_Tests.cpp
   Settings_Tests.cpp
-  Timeline_Tests.cpp )
+  Timeline_Tests.cpp)
+
+##########  SET OPENCV RELATED TEST FILES  ###############
+set(OPENSHOT_CV_TEST_FILES
+  CVTracker_Tests.cpp
+  CVStabilizer_Tests.cpp
+  # CVObjectDetection_Tests.cpp
+  )
 
 ################ TESTER EXECUTABLE #################
 # Create unit test executable (openshot-test)
 message (STATUS "Tests enabled, test executable will be built as tests/openshot-test")
-add_executable(openshot-test
-  tests.cpp
-  ${OPENSHOT_TEST_FILES} )
+if (OpenCV_FOUND)
+  add_executable(openshot-test
+    tests.cpp
+    ${OPENSHOT_TEST_FILES} 
+    ${OPENSHOT_CV_TEST_FILES})
+  
+  # Link libraries to the new executable
+  target_link_libraries(openshot-test openshot ${UnitTest++_LIBRARIES} ${OpenCV_LIBS} ${PROTOBUF_LIBRARY})
```

Comment:
```suggestion
# Link libraries to the new executable
target_link_libraries(openshot-test
  openshot
  ${UnitTest++_LIBRARIES}
  ${OPENSHOT_CV_LIBRARIES}
  )
```
Again, unconditionally. Really, I'll be wanting to get you set up with targets for all of this stuff (for one thing, that'll make it exportable), but for now this will serve.

Reply:
The other option is to do the configuration parts in multiple stages, like so:
```cmake
add_executable(openshot-test
  tests.cpp
  ${OPENSHOT_TEST_FILES}
  )
target_link_libraries(openshot-test openshot)

if(OpenCV_FOUND)
  target_sources(openshot-test
    CVTracker_Tests.cpp
    CVStabilizer_Tests.cpp
    # CVObjectDetection_Tests.cpp
    )
  target_link_libraries(openshot-test PUBLIC
    ${OpenCV_LIBS}
    ${PROTOBUF_LIBRARY}
    )
endif()
```
That also works perfectly fine, in fact it's slightly cleaner IMHO.